lib/spack/spack/spec.py

```diff
@@ -4878,7 +4887,12 @@ def from_node_dict(cls, node):
                 for val in values:
                     spec.compiler_flags.add_flag(name, val, False)
             else:
-                spec.variants[name] = vt.MultiValuedVariant.from_node_dict(name, values)
+                if isinstance(values, PropagateValue):
+                    spec.variants[name] = vt.AbstractVariant.from_node_dict(
+                        name, values.value, values.propagate
+                    )
+                else:
+                    spec.variants[name] = vt.AbstractVariant.from_node_dict(name, values)
```

Comment:
Why are we reconstructing `AbstractVariant`s instead of `MultivaluedVariant`s?  The semantic is different, and this function is used to read from JSON files (where specs are concrete).

Reply:
@alalazo  I thought that we should reconstruct `AbstractVariant` since the variants might not be `MultivaluedVariant`s, but if it's for semantic reasons I'll revert it.